frontend/modules/groups/groups.sagas.ts

```diff
@@ -148,34 +139,25 @@ export function* selectGroup({ group = {} }) {
 
 export function* addColorOverride({ groups = [], renderOnly }) {
 	try {
-		const TreeService = getAngularService('TreeService') as any;
 		const overridedToAdd = {};
 
 		for (let i = 0; i < groups.length; i++) {
 			const group = groups[i];
 			const color = hexToGLColor(group.color);
-			const treeMap = yield TreeService.getMap();
-
-			if (treeMap) {
-				const nodes = yield TreeService.getNodesFromSharedIds(group.objects);
-
-				if (nodes) {
-					const filteredNodes = nodes.filter((n) => n !== undefined);
-					const modelsMap = yield TreeService.getMeshMapFromNodes(filteredNodes);
-					const modelsList = Object.keys(modelsMap);
-
-					for (let j = 0; j < modelsList.length; j++) {
-						const modelKey = modelsList[j];
-						const meshIds = modelsMap[modelKey].meshes;
-						const [account, model] = modelKey.split('@');
-						Viewer.overrideMeshColor(account, model, meshIds, color);
-					}
-
-					const colorOverride = { models: modelsMap, color, id: group._id };
-					overridedToAdd[group._id] = colorOverride;
-				}
+			const nodesIds = yield select(getSelectNodesIdsFromSharedIds(group.objects));
+
+			const nodes = yield select(getSelectNodesByIds(nodesIds));
+			const filteredNodes = nodes.filter((n) => n !== undefined);
+			const modelsMap = yield select(getSelectMeshesByNodes(filteredNodes));
```

Comment:
> Instead of calling
> - getSelectNodesIdsFromSharedIDs
> - getSelectNodesByIds
> - getSelectMeshesByNodes
> 
> Can we make a function to just getMeshesBySharedIds?

Do you mean merging those into one selector?

Reply:
kind of. Yes make a new selector, but I don't think you need to do `getSelectNodesByIds` because you shouldn't need the node object to get mesh via `idToMeshes` mapping.

Reply:
How the new selector should work? How can I distinguish what it should return (meshes vs nodes)? I think it over complicates the logic. ;)

Yes, I need these nodes to get the `modelId` and `teamspace`. You wanted us to create something like namespace (`${teamspace}@${modeId}`) to handle connection between nodes and meshes (I think it was implemented this way before), so we did it, but now we have to get somehow this information. ;)
I suppose your idea makes sense only if we get rid of the namespace thing, and use model's id instead of it.
Please look at `getSelectMeshesByIds` in `tree.selectors.ts`.